Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -242,6 +242,12 @@ public override void Reset()
         /// <inheritdoc />
         public override void Disable()
         {
+            // Input System adds a gaze provider component on the main camera, which needs to be removed when the input system is disabled/removed.
+            // Otherwise the component would keep references to dead objects.
+            // Unity's way to remove component is to destroy it. Immediate is important, otherwise order of deinitialization will be arbitrary.
+            MixedRealityInputSystemProfile profile = ConfigurationProfile as MixedRealityInputSystemProfile;
+            UnityEngine.Object.DestroyImmediate(CameraCache.Main.gameObject.GetComponent(profile.PointerProfile.GazeProviderType.Type));
```

Comment:
Also, are we not able to destroy using the input system's `GazeProvider` reference? Probably after a cast to `MonoBehaviour` or `UnityEngine.Object`.

Reply:
I apprectiate both comments, because they helped me to find another bug, which I "fixed" by adding `Disable` calls into `Destroy` calls :smile: 

Also I didn't notice, that there was actually a call to DestroyImmediate here before, and I just needed to add one for play mode.

---
In reply to: [287528647](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4498#discussion_r287528647) [](ancestors = 287528647)